resources/pdns_authoritative_service_debian_sysvinit.rb

```diff
@@ -0,0 +1,89 @@
+#
+# Cookbook Name:: pdns
+# Resources:: pdns_authoritative_service
+#
+# Copyright 2017, Aetrion, LLC DBA DNSimple
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+resource_name :pdns_authoritative_service_debian_sysvinit
+
+provides :pdns_authoritative_service_sysvinit
+
+provides :pdns_authoritative_service, platform: 'ubuntu' do |node|
+  node['platform_version'].to_f >= 14.04
+end
+
+provides :pdns_authoritative_service, platform: 'debian' do |node|
+  node['platform_version'].to_i >= 8
+end
+
+property :instance_name, String, name_property: true
+property :cookbook, [String,nil], default: 'pdns'
```

Comment:
I would just make this a String type with the default. Can't really see a nil use case here.